src/nodestreamidguard.hpp

```diff
@@ -20,32 +20,41 @@
 
 #include <spdlog/spdlog.h>
 
+#include "metric.hpp"
+#include "model_metric_reporter.hpp"
 #include "ovinferrequestsqueue.hpp"
 #include "profiler.hpp"
 
 namespace ovms {
 
 struct NodeStreamIdGuard {
-    NodeStreamIdGuard(ovms::OVInferRequestsQueue& inferRequestsQueue) :
+    NodeStreamIdGuard(ovms::OVInferRequestsQueue& inferRequestsQueue, ModelMetricReporter& reporter) :
         inferRequestsQueue_(inferRequestsQueue),
-        futureStreamId(inferRequestsQueue_.getIdleStream()) {}
+        futureStreamId(inferRequestsQueue_.getIdleStream()),
+        reporter(reporter) {
+        INCREMENT_IF_ENABLED(reporter.currentRequests);
+    }
 
     ~NodeStreamIdGuard() {
         if (!disarmed) {
             if (!streamId) {
                 SPDLOG_DEBUG("Trying to disarm stream Id that is not needed anymore...");
                 streamId = futureStreamId.get();
+                INCREMENT_IF_ENABLED(reporter.inferReqActive);
```

Comment:
why increment?

Reply:
It was resolved that active infer req will always be a little bit off